keps/sig-storage/2848-transfer-volumesnapshots-namespaces/README.md

```diff
@@ -0,0 +1,795 @@
+bogged down.
+-->
+Users should be able to transfer VolumeSnapshots that reference the same VolumeSnapshotContent
+across namespaces. This allows users to quickly share snapshots, utilizing data for testing
+or backups in separate namespaces.
+#### Story 1
+
+- The `prod` namespace contains a PVC that holds user information. VolumeSnapshots are taken of
+this PVC on a regular basis.
+
+- An existing VolumeSnapshot can be transferred to a test namespace. The user initiates the request
+by creating a `VolumeSnapshotTransferRequest` in the source namespace.
+
+- Once created, a `VolumeSnapshotTransferApproval` resource in the target namespace will be created.
+If matching resources are found, the VolumeSnapshot controller can init the transfer. 
+
+- When the transfer is successful, the original VolumeSnapshot should be deleted to assure there is one-to-one mapping between VolumeSnapshot and VolumeSnapshotContent.
+
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+This KEP proposes the minimal to transfer VolumeSnapshot between namespaces, while avoiding breaking the current API.
+Many caveats have been discussed, including but not limited to :- 
+
+- [StorageClass Secrets](https://kubernetes-csi.github.io/docs/secrets-and-credentials-storage-class.html).
+- VolumeSnapshotContent.Spec.VolumeSnapshotRef is `immutable`. 
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+A `VolumeSnapshotTransferRequest` named `r` and `VolumeSnapshotTransferApproval` `a` are matched if all of the following conditions are met:
+
+- `r.metadata.name == a.spec.requestName`
+- `r.metadata.namespace == a.spec.targetNamespace`
+- `r.spec.source.namespace == a.metadata.namespace`
+- `r.spec.source.name == a.spec.source.name`
+- `r.spec.source.kind == a.spec.source.kind`
+- `r.spec.approvalName == a.metadata.name`
+
+Once matching is found, the transfer process is initiated by the controller, and it performs the following:
+
+- Creates a VolumeSnapshot in the target namespace.
+- Binds the VolumeSnapshotContent to the newly created VolumeSnapshot.
+- Deletes the source VolumeSnapshot.
+
```

Comment:
Please add descriptions on how to avoid issues caused by multiple `VolumeSnapshotTransferRequest`s for one source `VolumeSnapshot`. Unless transfer operations are atomic, two transition states for different `VolumeSnapshotTransferRequest`s will make undesirable state.

Adding status fields to store information on which `VolumeSnapshotTransferRequest` is handled currently will help block other request to be processed at the same time.

Reply:
shall we use what @jsafrane suggested here https://github.com/kubernetes/enhancements/pull/2849#discussion_r684995893

Reply:
shall we use what @jsafrane suggested here https://github.com/kubernetes/enhancements/pull/2849#discussion_r684995893